src/home/fetch-github/fetch-issues-preview.ts

```diff
@@ -97,12 +92,44 @@ export async function fetchIssuePreviews(): Promise<TaskNoFull[]> {
 
   return tasks;
 }
-function automaticLogin(error: unknown) {
-  const resetTime = error.response.headers["x-ratelimit-reset"];
-  const resetParsed = new Date(resetTime * 1000).toLocaleTimeString();
-
-  displayPopupMessage(
-    `GitHub API rate limit exceeded.`,
-    `You have been rate limited. Please log in to GitHub to increase your GitHub API limits, otherwise you can try again at ${resetParsed}.`
-  );
+
+function rateLimitModal(message: string) {
+  displayPopupMessage(`GitHub API rate limit exceeded.`, message);
+}
+
+type RateLimit = {
+  reset: number | null;
+  user: boolean;
+};
+
+export async function handleRateLimit(octokit?: Octokit, error?: unknown) {
+  const rate: RateLimit = {
+    reset: null,
+    user: false,
+  };
+
+  if (error) {
+    rate.reset = error.response.headers["x-ratelimit-reset"];
+  }
+
+  if (octokit) {
+    try {
+      const core = await octokit.rest.rateLimit.get();
+      const remaining = core.data.resources.core.remaining;
+      const reset = core.data.resources.core.reset;
+
+      rate.reset = !rate.reset && remaining === 0 ? reset : rate.reset;
+      rate.user = (await getGitHubUser()) ? true : false;
+    } catch (err) {
+      console.error("Error handling GitHub rate limit", err);
+    }
+  }
+
+  const resetParsed = rate.reset && new Date(rate.reset * 1000).toLocaleTimeString();
+
+  if (!rate.user) {
+    rateLimitModal(`You have been rate limited. Please log in to GitHub to increase your GitHub API limits, otherwise you can try again at ${resetParsed}.`);
```

Comment:
Does this take the UTC into account when displayed?

Reply:
> .toLocaleTimeString();

It uses local time. This is relevant for the user, and in my testing, it is accurate.

Reply:
- https://github.com/ubiquity/work.ubq.fi/blame/ecb2187ddc2067e99ae806af3430c67bf19a8d9d/src/home/fetch-github/fetch-issues-preview.ts#L100

The code is similar I guess but this pr is up to date with development if that's where prod is running from